freqtrade/rpc/api_server.py

```diff
@@ -0,0 +1,339 @@
+        """
+        pass
+
+    def rest_dump(self, return_value):
+        """ Helper function to jsonify object for a webserver """
+        return jsonify(return_value)
+
+    def rest_error(self, error_msg):
+        return jsonify({"error": error_msg}), 502
+
+    def register_rest_other(self):
+        """
+        Registers flask app URLs that are not calls to functionality in rpc.rpc.
+        :return:
+        """
+        self.app.register_error_handler(404, self.page_not_found)
+
+    def register_rest_rpc_urls(self):
+        """
+        Registers flask app URLs that are calls to functonality in rpc.rpc.
+
+        First two arguments passed are /URL and 'Label'
+        Label can be used as a shortcut when refactoring
+        :return:
+        """
+        # Actions to control the bot
+        self.app.add_url_rule(f'{BASE_URI}/start', 'start',
+                              view_func=self._start, methods=['POST'])
+        self.app.add_url_rule(f'{BASE_URI}/stop', 'stop', view_func=self._stop, methods=['POST'])
+        self.app.add_url_rule(f'{BASE_URI}/stopbuy', 'stopbuy',
+                              view_func=self._stopbuy, methods=['POST'])
+        self.app.add_url_rule(f'{BASE_URI}/reload_conf', 'reload_conf',
+                              view_func=self._reload_conf, methods=['POST'])
+        # Info commands
+        self.app.add_url_rule(f'{BASE_URI}/balance', 'balance',
+                              view_func=self._balance, methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/count', 'count', view_func=self._count, methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/daily', 'daily', view_func=self._daily, methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/edge', 'edge', view_func=self._edge, methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/profit', 'profit',
+                              view_func=self._profit, methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/performance', 'performance',
+                              view_func=self._performance, methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/status', 'status',
+                              view_func=self._status, methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/version', 'version',
+                              view_func=self._version, methods=['GET'])
+
+        # Combined actions and infos
+        self.app.add_url_rule(f'{BASE_URI}/blacklist', 'blacklist', view_func=self._blacklist,
+                              methods=['GET', 'POST'])
+        self.app.add_url_rule(f'{BASE_URI}/whitelist', 'whitelist', view_func=self._whitelist,
+                              methods=['GET'])
+        self.app.add_url_rule(f'{BASE_URI}/forcebuy', 'forcebuy',
+                              view_func=self._forcebuy, methods=['POST'])
+        self.app.add_url_rule(f'{BASE_URI}/forcesell', 'forcesell', view_func=self._forcesell,
+                              methods=['POST'])
+
```

Comment:
hmmm, if you had defined these endpoints in OpenAPI (swagger) and used `connexion` then you wouldn't need all these rules.

Reply:
i know that would sound good - however i did try connexion and i could not get it to call methods within a class instance - so it didn't have access to the running bot instance.